src/Farmer/Arm/App.fs

```diff
@@ -313,6 +315,7 @@ type ContainerApp =
                                             enabled = true
                                             appId = settings.AppId
                                             appPort = settings.Port |> Option.toNullable
+                                            appProtocol = settings.Protocol
```

Comment:
With `Protocol` now a union rather than a string, this code will need a pattern match to emit the correct string e.g.

```fsharp
match settings.Protocol with Http -> "http" | Gprc -> "grpc"
```

etc.